docs/parse_argparse.py

```diff
@@ -0,0 +1,118 @@
+            for line in fh_read:  # This keeps reading the file
+                if line.strip() == 'args = parser.parse_args()':
+                    line = line.strip()
+                    fh_write.write(line + '\n')
+                    break
+                line = line.strip()
+                fh_write.write(line + '\n')
+            fh_write.close()
+        
+        basename = os.path.basename(dir_name) #Extract the filename, eg: kmers.py
+        #print (basename)
+        len_base = len(basename) #needed to get the number of '=' in the header
+        file_name_base = os.path.splitext(basename)[0] #extract the kmers of kmers.py
+        file_name = file_name_base + ".rst" # make it kmers.rst   
+        #print(file_name)
+        main_path = os.path.dirname(os.path.abspath(dir_name)) # extract '/home/siddharth/Autometa/autometa/common/external'
+        #print(main_path)
+        path = main_path.split('/')
+        #print(main_path)
+        main_path = path [-1] # extraxt 'external'
+        path[path.index('autometa')] = 'docs/source/scripts_2' #replace autometa with the specified path in the list
+        path_ori=path='/'.join(path)
+
+        if not os.path.exists(path): #path now has 'docs/source/scripts_2/comon/external' instead of autometa
+            #make a new directory as per the structure in autometa
+            os.makedirs(path) #take care now, this will make each folder start with a lower case letter
+        path = os.path.join(path, file_name) #adding the kmers.rst to the above complete path
+        #now the path is 'docs/source/scripts_2/comon/external/samtools.rst'
+        #print(path)
+
+        cmd = 'python3 '+  os.path.abspath(path_temp_write) + ' -h'
+        #print(cmd)
+
+        path_temp_write_rst = os.path.abspath(path_temp_write)
+        #print(path_temp_write_rst)
+        base_path_temp_write = os.path.splitext(os.path.abspath(path_temp_write))[0]
+        #print(base_path_temp_write)
+        path_temp_write_rst = base_path_temp_write + '.rst'
+        #print(path_temp_write_rst)
+
+        #this block writes the argparse output to the respective '.rst' file with proper indentation
+        #this is the output that will be under '..code-block::' and will be displayed in html
+        with open(path_temp_write_rst, 'w+') as stdout:
+            subprocess.call(cmd, stdout=stdout, shell=True) #capture the argparse output
+            stdout.seek(0) #takes the cursor to beginning. Copy and indent stdout.
+            #The indentation is done because the arparse text needs to be indented
+            #under the ..code-block:: section in the final rst file
+            wrapper = textwrap.TextWrapper(initial_indent='\t', subsequent_indent='\t', width = 700)
+            wrapped=""
+            with open(path,'w') as fh_rst:
+                fh_rst.write('='*len_base + '\n' + basename + '\n' + '='*len_base + '\n' + '\n'+ '.. code-block:: shell' + '\n \n')
```

Comment:
You could define the output line first as a variable then write the variable making this more readable. Similarly, you've used the `str.join` method previously in this script. This would be the case where you could join these sections with '\n' using this aforementioned method.

e.g. 
```python
line = '\n'.join([your, output, line, sections])
fh_rst.write(line)
```